From ed6ef04f97d1490752ec03ad64de5e03c219a84e Mon Sep 17 00:00:00 2001
From: Camillo Bruni <cbruni@chromium.org>
Date: Wed, 1 Feb 2023 19:19:19 +0000
Subject: [PATCH] [blink] Optimise Node::ClearNeedsStyleRecalc

Move the HasRareData check up in the function to potentially skip
the the DynamicTo<Element> checks.

Bug: 808503
Change-Id: If9845a9a570cd0dd701bcfcc01a33e79da05569d
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4210256
Auto-Submit: Camillo Bruni <cbruni@chromium.org>
Commit-Queue: Xiaocheng Hu <xiaochengh@chromium.org>
Reviewed-by: Xiaocheng Hu <xiaochengh@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1099990}
---
 third_party/blink/renderer/core/dom/node.cc | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/third_party/blink/renderer/core/dom/node.cc
+++ b/third_party/blink/renderer/core/dom/node.cc
@@ -1374,10 +1374,12 @@
 void Node::ClearNeedsStyleRecalc() {
   node_flags_ &= ~kStyleChangeMask;
   ClearFlag(kForceReattachLayoutTree);
-
-  auto* element = DynamicTo<Element>(this);
-  if (element && HasRareData())
+  if (!HasRareData()) {
+    return;
+  }
+  if (auto* element = DynamicTo<Element>(this)) {
     element->SetAnimationStyleChange(false);
+  }
 }
 
 bool Node::InActiveDocument() const {
